reservations/flightreservation/flightreservation.cpp

```diff
@@ -55,22 +61,61 @@ void FlightReservation::start()
 
-    QString departureAirportName = reservationForMap["departureAirport"].toMap().value("name").toString();
-    QString departureAirportCode = reservationForMap["departureAirport"].toMap().value("iataCode").toString();
-    QDateTime departureTime = reservationForMap["departureTime"].toDateTime();
+    m_departureAirportName = reservationForMap["departureAirport"].toMap().value("name").toString();
+    m_departureAirportCode = reservationForMap["departureAirport"].toMap().value("iataCode").toString();
+    m_departureTime = reservationForMap["departureTime"].toDateTime();
 
-    QString arrivalAirportName = reservationForMap["arrivalAirport"].toMap().value("name").toString();
-    QString arrivalAirportCode = reservationForMap["arrivalAirport"].toMap().value("iataCode").toString();
-    QDateTime arrivalTime = reservationForMap["arrivalTime"].toDateTime();
+    m_arrivalAirportName = reservationForMap["arrivalAirport"].toMap().value("name").toString();
+    m_arrivalAirportCode = reservationForMap["arrivalAirport"].toMap().value("iataCode").toString();
+    m_arrivalTime = reservationForMap["arrivalTime"].toDateTime();
+
+    emit extractedData();
 }
 
-#include "flightreservation.moc"
+void FlightReservation::cacheData()
+{
+    QSqlDatabase db = QSqlDatabase::addDatabase("QSQLITE", "flight");
+    db.setDatabaseName("kdenowdb");
+
+    if (!db.open()) {
+        qWarning() << "Unable to open database";
+        qWarning() << db.lastError();
+    }
+    else {
+        qDebug() << "Database opened successfully";
+    }
+
+    QSqlQuery addQuery(db);
+    QString queryString = "create table if not exists Flight(id integer primary key autoincrement, reservationNumber varchar, name varchar, flight varchar, departureAirportName varchar, departureAirportCode varchar, departureTime varchar, arrivalAirportName varchar, arrivalAirportCode varchar, arrivalTime varchar)";
 
+    if (!addQuery.exec(queryString)) {
+        qWarning() << "Unable to create table";
+        qWarning() << addQuery.lastError();
+    }
+    else {
+        qDebug() << "Opened/Created successfully";
+    }
+
+    QSqlQuery updateQuery(db);
+    queryString = "insert into Flight values (:id, :reservationNumber, :name, :flight, :departureAirportName, :departureAirportCode, :departureTime, :arrivalAirportName, :arrivalAirportCode, :arrivalTime)";
+    updateQuery.prepare(queryString);
+    updateQuery.bindValue(":reservationNumber", m_reservationNumber);
+    updateQuery.bindValue(":name", m_name);
+    updateQuery.bindValue(":flight", m_flight);
+    updateQuery.bindValue(":departureAirportName", m_departureAirportName);
+    updateQuery.bindValue(":departureAirportCode", m_departureAirportCode);
+    updateQuery.bindValue(":departureTime", m_departureTime);
+    updateQuery.bindValue(":arrivalAirportName", m_arrivalAirportName);
+    updateQuery.bindValue(":arrivalAirportCode", m_arrivalAirportCode);
+    updateQuery.bindValue(":arrivalTime", m_arrivalTime);
```

Comment:
You are binding QDateTime with a varchar. It will produce error I guess. Please test it.